prime-dotnet-webapi/Services/EnrolleeService.cs

```diff
@@ -824,11 +833,31 @@ public async Task UpdateEnrolleeAdjudicator(int enrolleeId, int? adminId = null)
 
         public async Task<IEnumerable<BusinessEvent>> GetEnrolleeBusinessEventsAsync(int enrolleeId, IEnumerable<int> businessEventTypeCodes)
         {
-            return await _context.BusinessEvents
-                .Include(e => e.Admin)
-                .Where(e => e.EnrolleeId == enrolleeId && businessEventTypeCodes.Any(c => c == e.BusinessEventTypeCode))
-                .OrderByDescending(e => e.EventDate)
-                .ToListAsync();
+            var linkedPaperEnrolleeId = await GetLinkedPaperEnrolleeId(enrolleeId);
+            if (linkedPaperEnrolleeId != null)
+            {
+                return await _context.BusinessEvents
+                    .Include(e => e.Admin)
+                    .Where(e => e.EnrolleeId == enrolleeId || e.EnrolleeId == linkedPaperEnrolleeId && businessEventTypeCodes.Any(c => c == e.BusinessEventTypeCode))
```

Comment:
be very careful when mixing `||` and `&&` like this: the `&&` operator has higher precedence than the `||` operator. So this evaluates to 
e.EnrolleeId == enrolleeId 
OR 
e.EnrolleeId == linkedPaperEnrolleeId AND CodeFilter

I assume what you want is
```C#
.Where(e => (e.EnrolleeId == enrolleeId || e.EnrolleeId == linkedPaperEnrolleeId)
    && businessEventTypeCodes.Any(c => c == e.BusinessEventTypeCode))
```